test/controllers/MessagesControllerSpec.scala

```diff
@@ -371,6 +374,65 @@ class MessagesControllerSpec extends SpecBase with ScalaCheckPropertyChecks with
           status(result) mustEqual BAD_REQUEST
         }
       }
+
+      "must return InternalServerError if there has been a rejection from EIS due to virus found or invalid token" in {
+        val mockArrivalMovementRepository = mock[ArrivalMovementRepository]
+        val mockLockRepository            = mock[LockRepository]
+        val mockSubmitMessageService      = mock[SubmitMessageService]
+
+        when(mockLockRepository.lock(any())).thenReturn(Future.successful(true))
+        when(mockLockRepository.unlock(any())).thenReturn(Future.successful(()))
+        when(mockArrivalMovementRepository.get(any())).thenReturn(Future.successful(Some(arrival)))
+
+        when(mockSubmitMessageService.submitMessage(any(), any(), any(), any())(any()))
+          .thenReturn(Future.successful(SubmissionProcessingResult.SubmissionFailureRejected(VirusFoundOrInvalidToken)))
+
+        val application = baseApplicationBuilder
+          .overrides(
+            bind[ArrivalMovementRepository].toInstance(mockArrivalMovementRepository),
+            bind[LockRepository].toInstance(mockLockRepository),
+            bind[SubmitMessageService].toInstance(mockSubmitMessageService)
+          )
+          .build()
+
+        running(application) {
+          val request = FakeRequest(POST, routes.MessagesController.post(arrival.arrivalId).url).withXmlBody(requestXmlBody)
+
+          val result = route(application, request).value
+
+          status(result) mustEqual INTERNAL_SERVER_ERROR
+        }
+      }
```

Comment:
See question in MovementsControllerSpec.scala

Reply:
See https://github.com/hmrc/transit-movements-trader-at-destination/pull/154#discussion_r505456552